### PR TITLE
Add unit test for /routes/brackets.js

### PR DESCRIPTION
**Pull request by Kwanghyuk-Kim**
- Implement basic test cases for /routes/brackets.js
- .gz compression condition is not covered yet

Signed-off-by: KwangHyuk Kim <hyuki.kim@samsung.com>

### CODE REVIEW

**Comment by HunseopJeong** [comment]
@Kwanghyuk-Kim,
It seems to fail unit test.

**Comment by Kwanghyuk-Kim** [comment]
It's strange that test was done locally without issue.

**Comment by HunseopJeong** [comment]
Travis result:
```

[0m  1) test /routes/brackets.js GET .js error should test error for .js:

      [31mUncaught AssertionError: expected 200 to equal 404[0m
      [32m+ expected[0m [31m- actual[0m

      [31m-200[0m
      [32m+404[0m
      [0m[90m
      at agent.get.end (test/bracketsTest.js:57:29)
      at Test.Request.callback (node_modules/chai-http/node_modules/superagent/lib/node/index.js:619:12)
      at IncomingMessage.<anonymous> (node_modules/chai-http/node_modules/superagent/lib/node/index.js:795:18)
      at endReadableNT (_stream_readable.js:974:12)
      at _combinedTickCallback (internal/process/next_tick.js:80:11)
      at process._tickDomainCallback (internal/process/next_tick.js:128:9)
[0m
[0m  2) test /routes/brackets.js GET projectid error should test error for projectid:

      [31mUncaught AssertionError: expected 200 to equal 404[0m
      [32m+ expected[0m [31m- actual[0m

      [31m-200[0m
      [32m+404[0m
      [0m[90m
      at agent.get.end (test/bracketsTest.js:68:29)
      at Test.Request.callback (node_modules/chai-http/node_modules/superagent/lib/node/index.js:619:12)
      at IncomingMessage.<anonymous> (node_modules/chai-http/node_modules/superagent/lib/node/index.js:795:18)
      at endReadableNT (_stream_readable.js:974:12)
      at _combinedTickCallback (internal/process/next_tick.js:80:11)
      at process._tickDomainCallback (internal/process/next_tick.js:128:9)
[0m
```
My test result:
```
  16 passing (2s)
  1 failing

  1) test /routes/brackets.js test /proxy should test /proxy:
     Uncaught Error: connect ECONNREFUSED 127.0.0.1:443
      at Object.exports._errnoException (util.js:1022:11)
      at exports._exceptionWithHostPort (util.js:1045:20)
      at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1087:14)
```

**Comment by HunseopJeong** [comment]
I will trigger to rebuild

**Comment by Kwanghyuk-Kim** [comment]
Without '/brackets/test.js' and '/brackets/1111111', response status code can not be 200. :(

does Server have test.js or 1111111 under /brackets path ?

**Comment by Kwanghyuk-Kim** [comment]
and for below result, did you use MacOS or ubuntu ?

> 1) test /routes/brackets.js test /proxy should test /proxy:
>     Uncaught Error: connect ECONNREFUSED 127.0.0.1:443
>      at Object.exports._errnoException (util.js:1022:11)
>      at exports._exceptionWithHostPort (util.js:1045:20)
>      at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1087:14)

**Comment by HunseopJeong** [comment]
Yes. That is the MacOs result.

**Comment by Kwanghyuk-Kim** [comment]
your errored port = 443 and it's just for https.
It's interesting, node test might use 3002 port AFAIR. :(

**Comment by lslachciak** [comment]
I'm getting also ECONNREFUSED 127.0.0.1:443
After change
-      const httpClient  = options.protocol === 'http' ? http : https;
+      const httpClient  = options.protocol === 'https' ? https : http;
all tests (26)  are passed.

**Comment by Kwanghyuk-Kim** [comment]
@lslachciak 
Thank you for your idea. :)
However, I need any idea to make two tests be passed on travis. :(

**Comment by Kwanghyuk-Kim** [comment]
There were some difference between local and travis.
- working behavior of isLoggedIn is different between local and travis.

In order to check whether 404 not found test case will be passed, new pr was applied.
However, after fixing of this login issue, there will be other failures related with proxy and other.

**Comment by HunseopJeong** [comment]
Is it fixed after #48? Could you rebase it?

**Comment by Kwanghyuk-Kim** [comment]
> Is it fixed after #48? Could you rebase it?

yes,
rebased.

File: test/bracketsTest.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            .get('/brackets/'+projectId)
+            .end((error,res) => {
+              res.status.should.equal(200);
+              done();
+            });
+        });
+    });
+  });
+
+/*
```
Comment (HunseopJeong) [inline test/bracketsTest.js]: Why do you comment here?

File: test/bracketsTest.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
+process.env.NODE_ENV = 'test';
+
+var chai     = require('chai');
+var chaiHttp = require('chai-http');
+var mongoose = require('mongoose');
+var Project = require('../models/project');
+var server   = require('../app.js');
+var should   = chai.should();
```
Comment (lslachciak) [inline test/bracketsTest.js]: is the "should" var used anywhere?

Comment (Kwanghyuk-Kim) [reply test/bracketsTest.js]: it was removed, thx.

File: test/bracketsTest.js (hunk 1 above)
Comment (Kwanghyuk-Kim) [reply test/bracketsTest.js]: Oops, it's just for debuggin on travis-ci.
I will recover it soon.

File: test/bracketsTest.js (hunk 2 above)
Comment (HunseopJeong) [reply test/bracketsTest.js]: It is called for chai library to allow chainable assertions.
As I know, you don't remove it :(
http://cahijs.com/guide/styles/

Comment (Kwanghyuk-Kim) [reply test/bracketsTest.js]: My conclusion,
- res.should is working without import of chai.should.
- and global.should was assigned by someting automatically.
So, for this test, chai.should does not seem to be necessary for res.should.

Comment (HunseopJeong) [reply test/bracketsTest.js]: No. We should import 'chai.should'.
I got errors when I tested each tests without import of chai.should.

Comment (HunseopJeong) [reply test/bracketsTest.js]: Hunseopui-Mac-mini:WATT EatACarrot$ ./node_modules/.bin/nyc --reporter=txt ./node_modules/.bin/mocha --timeout=1000 ./test/appserverTest.js 


Listening on port 3002
  test app.js
    test 404 error
      1) should return 404 error
    test port success : normal case
      2) should test normal port
    test port error : negative case
      3) should test negative port
    test port error : NaN case
      4) should test NaN port

Comment (Kwanghyuk-Kim) [reply test/bracketsTest.js]: issue occurs only for mac ?
for me no issues and for travis no issues too.
however, to support mac, 'chai should' can be added again.

Comment (HunseopJeong) [reply test/bracketsTest.js]: Not only for mac.
You can test it.
When running npm test, first test sets the global.should.
It makes It seem to be no problem when we remove chai.should.
Try only one test without chai.should then you can get errors.
```
./node_modules/.bin/nyc --reporter=txt ./node_modules/.bin/mocha --timeout=1000 ./test/appserverTest.js
```

Comment (Kwanghyuk-Kim) [reply test/bracketsTest.js]: something = other test case.
Ok, chai,should was added.

**Pull request merged**